### PR TITLE
Update User Search Options Doc

### PR DESCRIPTION
**Pull request by khorne3**
https://auth0-docs-content-pr-5279.herokuapp.com/docs/users/search

Add guidance to user search.

### CODE REVIEW

File: articles/users/search.md [hunk 1]
```diff
@@ -94,11 +113,74 @@ This endpoint is **eventually consistent**, and as such, we recommend that you u
...
+  "family_name": ""
+}
+```
+
+This endpoint is **immediately consistent**, and as such, we recommend that you use this endpoint for:
+
+* User searches run during the authentication/authorization process 
+* User searches run as part of the account linking process.
+
+We do **not** recommend using the Users by ID endpoint for searches returning multiple users.
```
Comment (cabralmartin) [inline articles/users/search.md]: Remove (Redundant). Get by ID can only return one result.

File: articles/users/search.md [hunk 2]
```diff
@@ -160,10 +242,7 @@ The Users by Email endpoint is immediately consistent, and as such, we recommend
 * User searches run during the authentication/authorization process 
 * User searches run as part of the account linking process.
 
-We do **not** recommend using the Users by Email endpoint for:
-
-* Searches involving user attributes
-* Searches returning multiple users
+We do **not** recommend using the Users by Email endpoint for searches returning multiple users.
```
Comment (cabralmartin) [inline articles/users/search.md]: Same as above.

File: articles/users/search.md [hunk 3]
```diff
@@ -27,6 +28,25 @@ In this document, we use the terms **eventually consistent** and **immediately c
...
+
+* Use an immediately consistent endpoint during authentication pipeline
+* Use exact match searches (with the `raw` subfield) whenever possible
+* Use a well-known schema for metadata:
+  * Use consistent data types for properties
+  * Avoid dynamic property names
+  * Avoid large schema sizes and deep structures
+  * Avoid storing data you do not need for authentication and authorization purposes
+
+Please **avoid**:
```
Comment (cabralmartin) [inline articles/users/search.md]: Lets remove this `Please avoid` section. If the product supports this it looks weird we ask the user to "avoid" using.

File: articles/users/search.md [hunk 4]
```diff
@@ -27,6 +28,25 @@ In this document, we use the terms **eventually consistent** and **immediately c
 
 * **Immediately consistent**: When you request information about a user (or a group of users), the response will reflect the results of all successful write operations, including those that occured shortly prior to your request.
 
+## General Principles
+
+When running user searches:
+
+* Use an immediately consistent endpoint during authentication pipeline
```
Comment (mpaktiti) [inline articles/users/search.md]: I don't get this sentence. What's `during authentication pipeline`? You probably mean something like authentication transaction?

File: articles/users/search.md [hunk 5]
```diff
@@ -94,11 +113,72 @@ This endpoint is **eventually consistent**, and as such, we recommend that you u
...
+	"headers": [{
+		"name": "Authorization",
+		"value": "Bearer YOUR_MGMT_API_ACCESS_TOKEN"
+	}]
+}
+```
+
+**Sample Response**
+
+Successful calls to the endpoint returns a JSON object similar to the following:
```
Comment (mpaktiti) [inline articles/users/search.md]: Typo: `Successful calls to the endpoint returns a JSON object`

File: articles/users/search.md [hunk 6]
```diff
@@ -94,11 +113,72 @@ This endpoint is **eventually consistent**, and as such, we recommend that you u
...
+  "logins_count": 0,
+  "blocked": false,
+  "given_name": "",
+  "family_name": ""
+}
+```
+
+This endpoint is **immediately consistent**, and as such, we recommend that you use this endpoint for:
+
+* User searches run during the authentication/authorization process
```
Comment (mpaktiti) [inline articles/users/search.md]: Here you use `authentication/authorization process`. It's clearer than `authentication pipeline`. I also think you could lose the `/authorization` and simplify more.

File: articles/users/search.md [hunk 7]
```diff
@@ -292,7 +367,7 @@ When retrieving users in Auth0, there are three different API endpoints you can
 | Back of the office processes, such as changing a user's display name | [Users](#users) |
 | Searches involving user attributes | [Users](#users) |
 | Searches returning multiple users | [Users](#users) |
-| Operations requiring immediate consistency | [Users by Email](#users-by-email) |
-| Actions requiring user search as part of the Authentication Pipeline | [Users by Email](#users-by-email) |
+| Operations requiring immediate consistency | [Users by ID](#users-by-id) or [Users by Email](#users-by-email) |
+| Actions requiring user search as part of the Authentication Pipeline | [Users by ID](#users-by-id) or [Users by Email](#users-by-email) |
```
Comment (mpaktiti) [inline articles/users/search.md]: Same comment on the `Authentication Pipeline`. Use a consistent term throughout the doc.

**Review by mpaktiti — CHANGES_REQUESTED** [review]
Looks good, left a couple of minor comments

**Pull request merged**